src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,61 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String BASIC_DELIMITER = ",|:";
+    private static final int EMPTY_NULL_RETURN_VALUE = 0;
+
+
+    public static int calculateSum(String strValue) {
+        if (isEmptyOrNull(strValue))
+            return EMPTY_NULL_RETURN_VALUE;
+
+        return split(StringsToInts(getStringValues(strValue, getMatcher(strValue))));
+    }
+
+    private static String getDelimiter(Matcher m) {
+        if (m.find()) {
+            return BASIC_DELIMITER + "|" + m.group(1);
+        }
+        return BASIC_DELIMITER;
+    }
+
+    private static String[] getStringValues(String strValue, Matcher m) {
+        String delimiter = getDelimiter(getMatcher(strValue));
+        if (m.find()) {
+            return m.group(2).split(delimiter);
+        }
+        return strValue.split(delimiter);
+    }
+
+    private static Matcher getMatcher(String strValue) {
+        return Pattern.compile("//(.)\n(.*)").matcher(strValue);
+    }
+
+    private static int split(int[] numbers) {
+        int result = 0;
+        for (int number : numbers) {
+            result += number;
+        }
+        return result;
+    }
+
+    private static int[] StringsToInts(String[] values) throws RuntimeException {
+        int[] numbers = new int[values.length];
+        for (int i = 0; i < values.length; i++) {
+            numbers[i] = Integer.parseInt(values[i]);
+            isNegative(numbers[i]);
+        }
+        return numbers;
+    }
+
+    private static void isNegative(int number) {
+        if (number < 0) {
+            throw new RuntimeException();
```

Comment:
RuntimeException 대신 적절한 표준예외를 사용하면 더 좋을 것 같습니다!
메시지도 꼭 예외에 담아주세요. 그래야 어떤 문제인지 확실하게 전달할 수 있습니다. 
```suggestion
            throw new IllegalArgumentException("양수만 입력해 주세요.");
```